pkg/handlers/v1/filter.go

```diff
@@ -0,0 +1,72 @@
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/asecurityteam/awsconfig-filterd/pkg/domain"
+	"github.com/asecurityteam/awsconfig-filterd/pkg/logs"
+)
+
+// ConfigNotification is the basic structure of all incoming AWS Config SNS events
+// (https://docs.aws.amazon.com/config/latest/developerguide/example-sns-notification.html).
+type ConfigNotification struct {
+	Message            string `json:"Message"`
+	Timestamp          string `json:"Timestamp"`
+	ProcessedTimestamp string `json:"ProcessedTimestamp"`
+	Type               string `json:"Type"`
+	MessageID          string `json:"MessageId"`
+	TopicArn           string `json:"TopicArn"`
+	Subject            string `json:"Subject"`
+	SignatureVersion   string `json:"SignatureVersion"`
+	Signature          string `json:"Signature"`
+	SigningCertURL     string `json:"SigningCertURL"`
+	UnsubscribeURL     string `json:"UnsubscribeURL"`
+}
+
+// ConfigEvent represents a single AWS Config Configuration Item.
+type ConfigEvent struct {
+	ConfigurationItem domain.ConfigurationItem `json:"configurationItem"`
+}
+
+// ConfigFilterHandler applies a filter to AWS Config events.
+type ConfigFilterHandler struct {
+	Filter domain.ConfigFilter
+	LogFn  domain.LogFn
+	StatFn domain.StatFn
+}
+
+// Handle accepts Config events, applies a filter, and returns the events that match.
+func (h *ConfigFilterHandler) Handle(ctx context.Context, in ConfigNotification) (ConfigNotification, error) {
+	logger := h.LogFn(ctx)
+	stater := h.StatFn(ctx)
+
+	var event ConfigEvent
+	e := json.Unmarshal([]byte(in.Message), &event)
+	if e != nil {
+		logger.Error(logs.InvalidInput{Reason: e.Error()})
+		return ConfigNotification{}, domain.ErrEventDiscarded{Reason: e.Error()}
```

Comment:
Similar to the other comments about the error. This results in a failed Lambda call which may or may not be retried depending on the Lambda configuration. What do you _intend_ for the behavior to be here?

Reply:
I'm not sure how the lambda semantics would map to retries or other behavior, but the intention would be that a failure to unmarshal the input would be a non-retryable error.

Reply:
I think the part that is confusing to me is that the same error type (`domain.ErrEventDiscarded`) is used to signal both a permanent unmarshal failure as well as successful behavior of the system when dropping events that are filtered out. Retries are configurable in a number of different ways but the actual error and success cases need distinct signals.

Reply:
I'll add a separate error type for this case